yb-voyager/test/utils/voyager_cmd_runner.go

```diff
@@ -0,0 +1,186 @@
+package testutils
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	testcontainers "github.com/yugabyte/yb-voyager/yb-voyager/test/containers"
+)
+
+type ExitCode int
+
+const (
+	ExitCodeSuccess ExitCode = 0
+	ExitCodeFailure ExitCode = 1
+)
+
+func (ec ExitCode) String() string {
+	switch ec {
+	case ExitCodeSuccess:
+		return "Success"
+	case ExitCodeFailure:
+		return "Failure"
+	default:
+		return fmt.Sprintf("Exitcode(%d)", int(ec))
+	}
+}
+
+// wrapper for exec.Cmd to capture stdout and stderr and to extend it as per our test framework needs
+type VoyagerCommandRunner struct {
+	// basic command information to be provided
+	container   testcontainers.TestContainer
```

Comment:
I am not sure if we should keep the container as part of this struct information as there could be multiple cases where this is not very useful -
1. In case we want to run an `export data` command runner with different source configs (like different dbs ), like create an instance of an export-data command runner and `Run` with different configs?
2. For the `get data-migration-report` type of command, which interacts with different containers.

Reply:
Rightnow i assumed all commands would need this.
But we can make container field of the class/struct optional in that case.
Wouldn't that solve the problem? 

>  create an instance of an export-data command runner and Run with different configs?
I am not sure if i got you but I think that is the specific usecase which will be cleanly covered by this class.
For every case you can have different runner objects with specific fields(like container)

Reply:
nvm it is already optional unless it is a source/target db command
```
if container == nil && (isSourceCmd(cmdName) || isTargetCmd(cmdName)) {
		log.Fatal("Container cannot be nil for source/target commands")
	}
```